src/object/schema_system_catalog_install.cpp

```diff
@@ -0,0 +1,5193 @@
+    }
+
+  error_code = au_change_owner (class_mop, Au_dba_user);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  return NO_ERROR;
+}
+
+/*
+ * boot_add_data_type :
+ *
+ * returns : NO_ERROR if all OK, ER_ status otherwise
+ *
+ *   class(IN) :
+ *
+ * Note:
+ *
+ */
+static int
+boot_add_data_type (MOP class_mop)
+{
+  DB_OBJECT *obj;
+  DB_VALUE val;
+  int i;
+
+  const char *names[DB_TYPE_LAST] =
+  {
+    "INTEGER", "FLOAT", "DOUBLE", "STRING", "OBJECT",
+    "SET", "MULTISET", "SEQUENCE", "ELO", "TIME",
+    "TIMESTAMP", "DATE", "MONETARY", NULL /* VARIABLE */, NULL /* SUB */,
+    NULL /* POINTER */, NULL /* ERROR */, "SHORT", NULL /* VOBJ */,
+    NULL /* OID */,
+    NULL /* VALUE */, "NUMERIC", "BIT", "VARBIT", "CHAR",
+    "NCHAR", "VARNCHAR", NULL /* RESULTSET */, NULL /* MIDXKEY */,
+    NULL /* TABLE */,
+    "BIGINT", "DATETIME",
+    "BLOB", "CLOB", "ENUM",
+    "TIMESTAMPTZ", "TIMESTAMPLTZ", "DATETIMETZ", "DATETIMELTZ",
+    "JSON"
+  };
+
+  for (i = 0; i < DB_TYPE_LAST; i++)
+    {
+
+      if (names[i] != NULL)
+	{
+	  obj = db_create_internal (class_mop);
+	  if (obj == NULL)
+	    {
+	      assert (er_errid () != NO_ERROR);
+	      return er_errid ();
+	    }
+
+	  db_make_int (&val, i + 1);
+	  db_put_internal (obj, "type_id", &val);
+
+	  db_make_varchar (&val, 16, names[i], strlen (names[i]), LANG_SYS_CODESET, LANG_SYS_COLLATION);
+	  db_put_internal (obj, "type_name", &val);
+	}
+    }
+
+  return NO_ERROR;
+}
+
+/*
+ * boot_define_data_type :
+ *
+ * returns : NO_ERROR if all OK, ER_ status otherwise
+ *
+ *   class(IN) :
+ */
+static int
+boot_define_data_type (MOP class_mop)
+{
+  SM_TEMPLATE *def;
+  int error_code = NO_ERROR;
+
+  def = smt_edit_class_mop (class_mop, AU_ALTER);
+
+  error_code = smt_add_attribute (def, "type_id", "integer", NULL);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  /* TODO : DB migration tool */
+  error_code = smt_add_attribute (def, "type_name", "varchar(16)", NULL);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  error_code = sm_update_class (def, NULL);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  if (locator_has_heap (class_mop) == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      return er_errid ();
+    }
+
+  error_code = au_change_owner (class_mop, Au_dba_user);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  error_code = boot_add_data_type (class_mop);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  return NO_ERROR;
+}
+
+/*
+ * boot_define_stored_procedure :
+ *
+ * returns : NO_ERROR if all OK, ER_ status otherwise
+ *
+ *   class(IN) :
+ */
+static int
+boot_define_stored_procedure (MOP class_mop)
+{
+  SM_TEMPLATE *def;
+  char args_string[64];
+  int error_code = NO_ERROR;
+  const char *index_col_names[2] = { "sp_name", NULL };
+
+  def = smt_edit_class_mop (class_mop, AU_ALTER);
```

Comment:
Why not check if def is NULL?

Reply:
I just moved the code from the legacy routines in boot_cl.c.